core/src/value/partial.rs

```diff
@@ -0,0 +1,873 @@
+            DicomDate::Year(y) => (*y as i32, 12, 31),
+            DicomDate::Month(y, m) => {
+                let d = {
+                    if m == &12 {
+                        NaiveDate::from_ymd(*y as i32 + 1, 1, 1)
+                    } else {
+                        NaiveDate::from_ymd(*y as i32, *m as u32 + 1, 1)
+                    }
+                    .signed_duration_since(NaiveDate::from_ymd(*y as i32, *m as u32, 1))
+                    .num_days()
+                };
+                (*y as i32, *m as u32, d as u32)
+            }
+            DicomDate::Day(y, m, d) => (*y as i32, *m as u32, *d as u32),
+        };
+        NaiveDate::from_ymd_opt(y, m, d).context(InvalidDate)
+    }
+}
+
+impl AsRange<NaiveTime> for DicomTime {
+    fn earliest(&self) -> Result<NaiveTime> {
+        let fr: u32;
+        let (h, m, s, f) = match self {
+            DicomTime::Hour(h) => (h, &0, &0, &0),
+            DicomTime::Minute(h, m) => (h, m, &0, &0),
+            DicomTime::Second(h, m, s) => (h, m, s, &0),
+            DicomTime::Fraction(h, m, s, f, fp) => {
+                fr = *f * u32::pow(10, 6 - <u32>::from(*fp));
+                (h, m, s, &fr)
+            }
+        };
+        NaiveTime::from_hms_micro_opt((*h).into(), (*m).into(), (*s).into(), *f)
+            .context(InvalidTime)
+    }
+    fn latest(&self) -> Result<NaiveTime> {
+        let fr: u32;
+        let (h, m, s, f) = match self {
+            DicomTime::Hour(h) => (h, &59, &59, &999_999),
+            DicomTime::Minute(h, m) => (h, m, &59, &999_999),
+            DicomTime::Second(h, m, s) => (h, m, s, &999_999),
+            DicomTime::Fraction(h, m, s, f, fp) => {
+                fr = (*f * u32::pow(10, 6 - u32::from(*fp))) + (u32::pow(10, 6 - u32::from(*fp)))
+                    - 1;
+                (h, m, s, &fr)
+            }
+        };
+        NaiveTime::from_hms_micro_opt((*h).into(), (*m).into(), (*s).into(), *f)
+            .context(InvalidTime)
+    }
+}
+
+impl AsRange<DateTime<FixedOffset>> for DicomDateTime {
+    fn earliest(&self) -> Result<DateTime<FixedOffset>> {
+        let date = self.date.earliest()?;
+        let time = match self.time {
+            Some(time) => time.earliest()?,
+            None => NaiveTime::from_hms(0, 0, 0),
+        };
+
+        self.offset
+            .from_utc_date(&date)
+            .and_time(time)
+            .context(InvalidDateTime)
+    }
+
+    fn latest(&self) -> Result<DateTime<FixedOffset>> {
+        let date = self.date.latest()?;
+        let time = match self.time {
+            Some(time) => time.latest()?,
+            None => NaiveTime::from_hms_micro(23, 59, 59, 999_999),
+        };
+        self.offset
+            .from_utc_date(&date)
+            .and_time(time)
+            .context(InvalidDateTime)
+    }
+}
+
+impl DicomDate {
+    /**
+     * Retrieves a `chrono::NaiveDate` if value is precise.
+     */
+    pub fn as_naive_date(self) -> Result<NaiveDate> {
```

Comment:
Revisiting [C-CONV](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv), `to_naive_date` would be more conventional (it's owned -> owned for `Copy` types). Same for `to_naive_time` and `to_chrono_datetime`.